ArtOfIllusion/src/artofillusion/Scene.java

```diff
@@ -537,25 +532,17 @@ public void addMaterial(Material mat)
   public void addMaterial(Material mat, int index)
   {
     materials.add(index, mat);
-    for (int i = 0; i < materialListeners.size(); i++)
-      materialListeners.elementAt(i).itemAdded(materials.size()-1, mat);
+    int size = materials.size()-1;
+    materialListeners.forEach(listener -> listener.itemAdded(size, mat));
   }
 
   /** Remove a Material from the scene. */
 
   public void removeMaterial(int which)
   {
-    Material mat = materials.elementAt(which);
-
-    materials.removeElementAt(which);
-    for (int i = 0; i < materialListeners.size(); i++)
-      materialListeners.elementAt(i).itemRemoved(which, mat);
-    for (int i = 0; i < objects.size(); i++)
-      {
-        ObjectInfo obj = objects.elementAt(i);
-        if (obj.getObject().getMaterial() == mat)
-          obj.setMaterial(null, null);
-      }
+    Material mat = materials.remove(which);
+    materialListeners.forEach(listener -> listener.itemRemoved(which, mat));
+    objects.stream().filter(item -> item.getObject().getMaterial() == mat).forEach(item -> item.setMaterial(null, null));
```

Comment:
This line is 117 characters long, not including the leading whitespace. That is too long.

For readability, This should be broken at the `.forEach(` at least, and I might also argue at the `.filter(` as well, though that I could see either way.

It can be aligned to match the `.stream(` (All of the periods in a neat column)